tinodesdk/src/main/java/co/tinode/tinodesdk/Topic.java

```diff
@@ -1947,11 +1970,11 @@ public MetaGetBuilder withLaterData(Integer limit) {
          * @param limit number of messages to fetch
          */
         public MetaGetBuilder withEarlierData(Integer limit) {
-            MsgRange r = topic.getCachedMessagesRange();
-            if (r == null) {
+            MsgRange r = topic.getMissingMessageRange();
+            if (r == null || r.low <= 1) {
                 return withData(null, null, limit);
             }
-            return withData(null, r.low > 1 ? r.low : null, limit);
+            return withData(null, r.low, limit);
```

Comment:
Wouldn't you want to fetch the missing range `r` and not messages before this range?
It should be more like `withData(null, r.hi, limit)`.

Reply:
Good point. I need to fix and test this. I'll update this PR tomorrow.

Reply:
Done